### PR TITLE
feat: generic Product Quantizatoin

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

File: rust/lance-arrow/src/floats.rs [hunk 1]
```diff
@@ -69,7 +69,17 @@ pub trait ArrowFloatType {
...
+    + Bounded
+    + Sum
+    + AddAssign<Self>
+    + AsPrimitive<f64>
+    + DivAssign
+    + Send
+    + Sync
+    + Copy
+    + One
+    + Zero
```
Comment (wjones127) [inline rust/lance-arrow/src/floats.rs:82]: I'm confused. `Float` already implies most of these doesn't it?

**Review by wjones127 — APPROVED** [review]
Got 1 minor question

File: rust/lance-arrow/src/floats.rs (hunk 1 above)
Comment (eddyxu) [reply rust/lance-arrow/src/floats.rs:82]: Hmm, yea maybe it actually asks me to include these trait from the caller side? lemme take a look

Comment (eddyxu) [reply rust/lance-arrow/src/floats.rs]: Removed Zero and One

**Pull request merged**